src/socket/address.cr

```diff
@@ -132,10 +132,6 @@ class Socket
       end
     end
 
-    private def address(addr) : Nil
-      # shouldn't happen
-    end
```

Comment:
How about keeping this method, and have it raise a `@addr is not set` exception, and remove the two `not_nil!` assertions above? I'm guessing it would be more descriptive should we land in that case.